CMake/kwiver-depends-VTK.cmake

```diff
@@ -6,16 +6,19 @@ option( KWIVER_ENABLE_VTK
   )
 
 if( KWIVER_ENABLE_VTK )
-    find_package(VTK REQUIRED
-        COMPONENTS
-        vtkCommonCore
-        vtkCommonDataModel
-        )
-    if(VTK_VERSION VERSION_LESS 8.2)
-        message(FATAL_ERROR "${PROJECT_NAME} supports VTK >= v8.2 "
-            "(Found ${VTK_VERSION})")
-    endif()
-
-  include(${VTK_USE_FILE})
-
+  find_package(VTK REQUIRED
```

Comment:
@mleotta One thing I don't like here is that the component labels passed to the `find_package` call have also changed starting with v8.9 (I think that's the version). Now we should pass CommonCore rather than vtkCommonCore. The old call is deprecated and still works but it does produce a CMake warning for every component added. Not a huge deal but I still hate seeing them. 

The alternative I can think of would be to add a `find_package( VTK 8.9 COMPONENTS CommonCore ...)` first. The if `VTK_FOUND` isn't set or false, we can drop down to a `find_package( VTK 8.8 COMPONENTS vtkCommonCore ...)`. It's a bit uglier in the CMake code but would eliminate the warnings and be more future proof in the event the deprecation turns to a failure. It's possible that there is a more slick way to handle it but I'm not thinking of anything at the moment.

Reply:
Alternatively, just have an option what VTK version to use, and only look for that version? We've done that sort of thing before e.g. for Qt. (For that matter, due to incompatibilities between 8.0 and 8.2, I believe ViViA has *had* to do that for those versions.)

This might also be important if both VTK 8 and VTK 9 can be found, but e.g. if planning to build ViViA, which requires 8.0.

Reply:
At the moment, the KWIVER use of VTK can use either 8 or 9, so I don't think we want to add an option in KWIVER for which VTK to look for.  It should be able to find and use either 8 or 9.  We have an option in Fletch for which VTK to build so that should be sufficient.  Downstream applications (ViViA, TeleSculptor, etc.) can have there own requirements about which VTK they need and push that request to Fletch in a superbuild.  KWIVER can be version agnostic, at least until there is code in KWIVER itself that requires on version over another.

Reply:
That's basically what I'm doing in kwiver-bootstrap as well. When building Vivia, we need to set Fletch to VTK 8.0 and disable VTK support in Kwiver, which works well. There isn't any combination of VTK versions that works for kwiver and vivia.

Reply:
We could probably drop the VTK requirement for KWIVER down to 8.0 if that would help.  I believe the things that we need 8.2 for are still in TeleSculptor and the things we have moved into KWIVER so far will be okay with 8.0.